localstack-core/localstack/services/apigateway/next_gen/execute_api/handlers/integration_response.py

```diff
@@ -68,12 +68,16 @@ def __call__(
         # here we update the `Response`. We basically need to remove all headers and replace them with the mapping, then
         # override them if there are overrides.
         # for the body, it will maybe depend on configuration?
+        for key in list(response.headers.keys(lower=True)):
+            if key not in ("connection", "content-type"):
+                del response.headers[key]
```

Comment:
I'm not too sure about the whole "reconstruct" the whole Response object from the integration response handler, it almost defeat the purpose a little bit (we could just create a dict with the integration "response", and then only serialize it in the next "request handler", instead of having the integration response handler recreating the response completely. But still the big philosophical question I don't have a proper answer for 😄 

But yes, sorry, we could basically override the whole `Response.headers` directly. One way would be initialize an empty `Headers` object and add the few values that should be kept, but this is a bit annoying because we need to be mindful of the fact that some headers might be "multi-values" and we need to call individually `getlist` on every key, and then update this with the mapping. 

Something like:
```python
response_headers = Headers()
for header in ("connection", "content-type"):
    if values := response.headers.getlist(header):
        for value in values:
            response_headers.add(header, value)
```
or maybe a simplified version:
```python
response_headers = Headers()
for header in ("connection", "content-type"):
    if values := response.headers.getlist(header):
        response_headers.setlist(header, values)
```

I don't really have a preference, it also look a bit big, and same level of indentation!

Reply:
@cloutierMat I think I'll merge the PR now, and we can update this in #11144 if you think any of the one above is better! I think we will need to rebase anyway. Is that ok?

Reply:
Yes, totally. Sorry for the delay, I don't think it is a deal breaker at all. Let's have that conversation and adjust in a future PR.

Reply:
No worries, no delay at all here 😄 Just didn't want to block future merge with conflicts 😅 sorry for that!